### DISCUSSION TITLE
kafka client 들의 내부 구현 언어는 무엇인가

### DISCUSSION THREAD

**Discussion by sejongjeong** [kyungjunleeme/kafka_study#23] [Q&A]
https://github.com/IBM/sarama

내부 구현 C?

**Comment by kyungjunleeme** [comment]
```
kafka-python은 Kafka의 네이티브 Java API를 Python으로 번역한 것이 아니라, Kafka 프로토콜(TCP 메시징 레이어)을 직접 구현한 Python 라이브러리입니다.
이 라이브러리는 Kafka의 주요 기능(프로듀싱, 컨슈밍, 메타데이터 관리 등)을 Python 스타일의 API로 제공하며, 이를 통해 Python 개발자도 Kafka의 강력한 메시징 시스템을 활용할 수 있도록 설계되었습니다.
Java API와 비교했을 때, Python 환경의 비동기 프로그래밍(예: asyncio)과 Pythonic한 문법 스타일을 더한 점이 큰 특징입니다.
```

### 주요 이점
- Kafka의 고유 기능을 Python에서 쉽게 접근 가능.
- Java를 몰라도 Python만으로도 Kafka 활용 가능.
- 커스텀 메시징 로직을 Python 애플리케이션에서 손쉽게 구현.
- 결론적으로, kafka-python은 Kafka의 프로토콜에 대한 철저한 이해를 바탕으로 만들어진 비공식 Python 클라이언트 라이브러리입니다. Java와 동일한 기능성을 Python에서 사용할 수 있게 하며, Kafka의 유연성과 강력함을 Python 환경에서도 누릴 수 있도록 지원합니다.

**Comment by kyungjunleeme** [comment]
<img width="671" alt="image" src="https://github.com/user-attachments/assets/5571fabb-a392-4a26-b3d6-67355663cf85">

https://chatgpt.com/share/6753f47f-545c-8009-bc87-0555b3e41d5f

**Comment by hellices** [comment]
[Go (AKA golang) client](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=30736606#Clients-Go(AKAgolang))
sarama는 순수 go 라이브러리라고 합니다.

**Reply by kyungjunleeme** [reply]
아하 그렇군요!
